### ISSUE TITLE
Spaces in root path cause plugins to fail

### ISSUE THREAD

**Issue by cemeceme** [thought-machine/please#2863]
If the current project being built contains spaces and it uses a plugin, `plugin_repo` will fail, as it cannot locate arcat.
i.e. if the project resides in `/dir/with spaces/proj` will fail, as `/dir/with` will get used instead of the full path to the tool.

This appears to be related to `$TOOL` not being in quotes on [this line](https://github.com/thought-machine/please/blob/c979fd781b6b528fc8af07f9e7e8d3bb9eeecceb/rules/subrepo_rules.build_defs#L347). This also goes for all 3 instances of the use of `$TOOL` in the `subrepo_rules.build_defs` file ([Here](https://github.com/thought-machine/please/blob/c979fd781b6b528fc8af07f9e7e8d3bb9eeecceb/rules/subrepo_rules.build_defs#L117), [here](https://github.com/thought-machine/please/blob/c979fd781b6b528fc8af07f9e7e8d3bb9eeecceb/rules/subrepo_rules.build_defs#L119), [here](https://github.com/thought-machine/please/blob/c979fd781b6b528fc8af07f9e7e8d3bb9eeecceb/rules/subrepo_rules.build_defs#L347)).

While just adding quotes around each `$TOOL` fixes this issue, I'm starting to wonder if a more universal solution could be offered by please? It looks like just passing quoted values into variables is not feasible ([as per this issue](https://github.com/thought-machine/please/issues/1215)), but some sort of mechanism to wrap strings like this properly with please would be very useful.

**Comment by Tatskaari** [comment]
Yeah, this is kind of a known issue. I don't have any major ideas. Using spaces around $TOOL might work, but it isn't going to work for e.g. $TOOLS where the spaces are used to separate the tool paths. I don't have any great ideas, however to suggest that you avoid using filepaths with spaces. 

Open to suggestion (or better yet, a PR) though

**Comment by cemeceme** [comment]
I was thinking more about this issue, and I was wondering: Wouldnt it work if please did the string substitutions instead?
For instance, after passing in the `cmd` argument for `build_rule`, please could just substitute out variables like `$TOOL` and `$SRCS` etc. It should have enough information before spinning up a shell to do so.

Now, in order to keep backwards compatibility, or to make sure this substitution can be disabled, the `cmd` argument for `build_rule` could accept two types: the current `str` type that wouldnt do the substitution and a new `cmdStr`(let me know if you think of a better name for this) type, that is an object produced by a `buildCmdStr(str)`(again, better names are appreciated) function.

With this, please could use a much saner substitution style (e.g. it appears python uses the `{variable}` style with `.format`), that would sidestep having to deal with bash.

I would be open to making a pull request with a prototype, but it might take a while since I havent written any Go before.

**Comment by alexeymuranov** [comment]
Curiously, unusual characters in the path name also break another popular Go build tool: https://github.com/go-task/task/issues/2874.
